### DISCUSSION TITLE
Is it possible to to filter by favourites in album/folder view?

### DISCUSSION THREAD

**Discussion by reborn-41273098** [photoprism/photoprism#2482] [Ideas]
Hi,

in normal search tab we can filter by favorite *photos* via `favorite:1`.
Can *albums* or *folders* (which are marked as favorites by star instead of heart icon), also be filtered by this or similar setting in their respective views?

For example: Select all albums, which start with `foo` and are marked as favorites?

**Comment by graciousgrey** [comment]
On /albums  and /folders you should be able to search for `favorite:true foo`

See: https://demo.photoprism.app/albums?q=favorite%3Atrue%20ber

**Reply by reborn-41273098** [reply]
Thanks for responding quickly @graciousgrey .
I guess what has confused me, is:
If I *don't* want favorites to show up, they still are displayed anyway: https://demo.photoprism.app/albums?q=favorite%3Afalse
IMO it would be more consistent to filter out the two favorites, if `favorite:false` explicitely is given.

**Reply by lastzero** [reply]
Other users may expect something different. At the moment it is a bool flag with 2 values, not 3 like true, false, both.

**Reply by reborn-41273098** [reply]
@lastzero what about following:
- If favorite *explicitly* given via form, filter albums/folders strictly by this flag
- If *not* - assuming photoprism UI doesn't set default value for favorite itself -, don't consider flag at all for filtering

Relevant code location might be https://github.com/photoprism/photoprism/blob/e8a1e85/internal/search/albums.go#L115-L117 .

with a change to handle `favorite:false` (sorry for lack of go lang, consider it as pseudo code):
```
if f.Favorite != "" {
  s = s.Where("albums.album_favorite = " + f.Favorite)
}
```

**Reply by lastzero** [reply]
Yeah, we'd have to change the Form fields from bool to string for this. That's possible, but we need to focus on other things first.
